compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5826,6 +5826,78 @@ TR::Register *OMR::Power::TreeEvaluator::retrieveTOCRegister(TR::Node *node, TR:
    return cg->machine()->getRealRegister(TR::RealRegister::gr2);
 }
 
+TR::Register *OMR::Power::TreeEvaluator::sbyteswapEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT(node->getNumChildren() == 1, "Wrong number of children in sbyteswapEvaluator");
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *tgtRegister = cg->allocateRegister();
+
+   TR::Node *firstNonConversionOpCodeNode = node->getFirstChild();
+   TR::DataType nodeType = firstNonConversionOpCodeNode->getType();
+
+   //Move through descendants until a non conversion opcode is reached,
+   //while making sure all nodes have a ref count of 1 and the types are between 2-8 bytes
+   while (firstNonConversionOpCodeNode->getOpCode().isConversion() &&
+          firstNonConversionOpCodeNode->getReferenceCount() == 1 &&
+          (nodeType.isInt16() || nodeType.isInt32() || nodeType.isInt64()))
+      {
+      firstNonConversionOpCodeNode = firstNonConversionOpCodeNode->getFirstChild();
+      nodeType = firstNonConversionOpCodeNode->getType();
+      }
+
+   if (!firstNonConversionOpCodeNode->getRegister() &&
+       firstNonConversionOpCodeNode->getOpCode().isMemoryReference() &&
+       firstNonConversionOpCodeNode->getReferenceCount() == 1 &&
+       (nodeType.isInt16() || nodeType.isInt32() || nodeType.isInt64()))
+      {
+      TR::MemoryReference *tempMR = new (cg->trHeapMemory()) TR::MemoryReference(firstNonConversionOpCodeNode, 2, cg);
```

Comment:
If #5536 goes in first we should switch to using a `::create*` API for this.

Reply:
Good point. That PR is very close to getting in now (and fixes a ppc32 build break), so it's probably best for this PR to wait on that first.

Reply:
Since that PR has now merged, I've rebased this PR and switched to using the new `MemoryReference` creation APIs.